gcs/conversions.go

```diff
@@ -124,26 +124,29 @@ func toObject(in *storagev1.Object) (out *Object, err error) {
 	}
 
 	// CRC32C
-	crc32cString, err := base64.StdEncoding.DecodeString(in.Crc32c)
-	if err != nil {
-		err = fmt.Errorf("Decoding Crc32c field: %v", err)
-		return
-	}
-
-	if len(crc32cString) != 4 {
-		err = fmt.Errorf(
-			"Wrong length for decoded Crc32c field: %d",
-			len(crc32cString))
+	// No CRC32C or MD5 values returned for CMEK buckets
+	if in.Crc32c != "" {
+		var crc32cString []byte
+		crc32cString, err = base64.StdEncoding.DecodeString(in.Crc32c)
+		if err != nil {
+			err = fmt.Errorf("Decoding Crc32c field: %v", err)
+			return
+		}
 
-		return
-	}
+		if len(crc32cString) != 4 {
+			err = fmt.Errorf(
+				"Wrong length for decoded Crc32c field: %d",
+				len(crc32cString))
 
-	out.CRC32C =
-		uint32(crc32cString[0])<<24 |
-			uint32(crc32cString[1])<<16 |
-			uint32(crc32cString[2])<<8 |
-			uint32(crc32cString[3])<<0
+			return
+		}
 
+		out.CRC32C =
```

Comment:
I'm uncomfortable with leaving this as zero when the server didn't specify it
-- that means downstream code will think it is invalid.

If it's truly expected that you don't get a CRC from the listing for these
objects (i.e. not a GCS bug), then the type of the field should be changed to
`*uint32` instead.

Reply:
OK - guess that depends on whether an extra roundtrip to populate MD5 / CRC is worth it.  Looks like MD5 already does this (for composite objects).

Reply:
Not sure what you mean about MD5; could you link to the code you're talking about?

Reply:
Sorry - I meant MD5 uses the same approach as you suggest:

https://github.com/jacobsa/gcloud/blob/9291bd1e83086329677b72de9dea5d77551ae057/gcs/object.go#L37

Reply:
Oh yes, exactly. So could you change CRC32C to be `*uint32`?

Reply:
Resolved in 9dfd009